2-mandatory/4-stocks.js

```diff
@@ -33,8 +33,16 @@ const CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS = [
         Solve the smaller problems, and then build those solutions back up to solve the larger problem.
         Functions can help with this!
 */
+
 function getAveragePrices(closingPricesForAllStocks) {
-    // TODO
+    return closingPricesForAllStocks.map((value, index) => {
+        let sum = 0, counter = 0;
+        for (let item of value) {
+            sum += item;
+            counter++;
+        }
+        return parseFloat((sum/counter).toFixed(2));
```

Comment:
Can you think of a way to achieve the same result without converting the number into a string and back? Converting a number to a string is a relatively expensive operation and should be avoided if possible.

Reply:
I have tried to doing that for one of the previous assignments, but alas failed to do so. It seems to me that without parseFloat() the 2 decimal places are cut if their value is 0. Eg. 50.00 --> 50.

What's the proper way to do it Chief? 😎

Reply:
It is correct to use `toFixed(2)` when the requirement is to have always 2 decimal digits, but that would convert the number into a string, and in this case the exercise wanted you to return a number instead.

To approximate the number `x` to 2 decimal digits, but without converting it to string and back, you can use the following formula: `Math.round(x * 100) / 100)`

Reply:
@gixo interesting 🤔 didn't know that.